lib/features/context-pad/ContextPadProvider.js

```diff
@@ -366,24 +366,24 @@ ContextPadProvider.prototype.getContextPadEntries = function(element) {
     'bpmn:FlowNode',
     'bpmn:InteractionNode',
     'bpmn:DataObjectReference',
-    'bpmn:DataStoreReference'
+    'bpmn:DataStoreReference',
+    'bpmn:TextAnnotation'
   ])) {
-
-    assign(actions, {
-      'append.text-annotation': appendAction('bpmn:TextAnnotation', 'bpmn-icon-text-annotation'),
-
+    var actionsMenu = {
       'connect': {
         group: 'connect',
         className: 'bpmn-icon-connection-multi',
         title: translate('Connect using ' +
-                  (businessObject.isForCompensation ? '' : 'Sequence/MessageFlow or ') +
-                  'Association'),
+          (businessObject.isForCompensation ? '' : 'Sequence/MessageFlow or ') +
+          'Association'),
         action: {
           click: startConnect,
           dragstart: startConnect
         }
       }
-    });
+    };
+    if (!is(businessObject, 'bpmn:TextAnnotation')) actionsMenu['append.text-annotation'] = appendAction('bpmn:TextAnnotation', 'bpmn-icon-text-annotation');
+    assign(actions, actionsMenu);
   }
```

Comment:
This is kind of doubled logic here. Why not simply:
```
  if (isAny(businessObject, [
    'bpmn:FlowNode',
    'bpmn:InteractionNode',
    'bpmn:DataObjectReference',
    'bpmn:DataStoreReference'
  ])) {

    assign(actions, {
      'append.text-annotation': appendAction('bpmn:TextAnnotation', 'bpmn-icon-text-annotation'),

      'connect': {
        group: 'connect',
        className: 'bpmn-icon-connection-multi',
        title: translate('Connect using ' +
                    (businessObject.isForCompensation ? '' : 'Sequence/MessageFlow or ') +
                    'Association'),
        action: {
          click: startConnect,
          dragstart: startConnect
        }
      }
    });
  }

  if (is(businessObject, 'bpmn:TextAnnotation')) {
    assign(actions, {
      'connect': {
        group: 'connect',
        className: 'bpmn-icon-connection-multi',
        title: translate('Connect using Association'),
        action: {
          click: startConnect,
          dragstart: startConnect
        }
      }
    });
  }
```

Reply:
hello @MaxTru thanks for your suggestions i have retested with and it works very fine 😄